### PR TITLE
707

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

File: crud/src/test/java/com/redhat/lightblue/mediator/CompositeFinderTest.java [hunk 1]
```diff
@@ -890,6 +890,28 @@ public void looping_entities_with_or_query_704() throws Exception {
...
+                          "                              ]"+
+                          "                   }"+
+                          "             }"+
+                          "          ]},"+
+                          "          {'field':'refchild.*.refparent.*.field2','op':'=','rvalue':true}"+
+                          "]}"));
+        fr.setProjection(projection("[{'field':'*'},{'field':'refchild'}]"));
+        fr.setEntityVersion(new EntityVersion("root_loop","1.0.0."));
+        Response response=mediator.find(fr);
+        System.out.println(response.getEntityData());
```
Comment (BVulaj) [inline crud/src/test/java/com/redhat/lightblue/mediator/CompositeFinderTest.java]: Anything we can assert here?

File: crud/src/main/java/com/redhat/lightblue/assoc/ep/Assemble.java [hunk 2]
```diff
@@ -102,8 +102,10 @@ public void initialize() {
             BatchAssembler batchAssembler = new BatchAssembler(256, aq, destination.getValue(), ctx);
             assemblers.add(ctx.getExecutor().submit(() -> {
                 if (aq.getQuery() == null) {
-                    results.stream().forEach(batchAssembler::addDoc);
-                    batchAssembler.endDoc();
+                    if(aq.isAlwaysTrue()) {
```
Comment (BVulaj) [inline crud/src/main/java/com/redhat/lightblue/assoc/ep/Assemble.java]: Maybe just make this a compound if?

Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/assoc/ep/Assemble.java]: No, the else applies to the outer if

File: crud/src/test/java/com/redhat/lightblue/mediator/CompositeFinderTest.java (hunk 1 above)
Comment (bserdar) [inline crud/src/test/java/com/redhat/lightblue/mediator/CompositeFinderTest.java]: Done

**Pull request merged**